### PR TITLE
SetServerSettings function

### PR DESCRIPTION
**Pull request by TacoError**

### CODE REVIEW

File: server/session/session.go [hunk 1]
```diff
@@ -538,3 +544,14 @@ func (s *Session) sendAvailableEntities(w *world.World) {
 	}
 	s.writePacket(&packet.AvailableActorIdentifiers{SerialisedEntityIdentifiers: serializedEntityData})
 }
+
+// SetServerSettings Sets the server settings data.
+// This will be used when a ServerSettingsRequest packet is
+// sent by the client.
+func (s *Session) SetServerSettings(form form.Form) {
+	h := s.handlers[packet.IDModalFormResponse].(*ModalFormResponseHandler)
+	h.forms[h.currentID.Add(1)] = form
```
Comment (Prim69) [inline server/session/session.go:553]: Inc()

File: server/session/handler_server_settings_request.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package session
+
+import (
+	"github.com/sandertv/gophertunnel/minecraft/protocol/packet"
+)
+
+// ServerSettingsRequest Is send by the client
```
Comment (JustTalDevelops) [inline server/session/handler_server_settings_request.go:7]: Would fix the grammar here, and also inline the comment below into the one.

File: server/session/handler_server_settings_request.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// in order to request a ServerSettingsResponse packet
+type ServerSettingsRequest struct{}
+
+// Handle Sends the ServerSettingsResponse packet.
```
Comment (JustTalDevelops) [inline server/session/handler_server_settings_request.go:11]: Can just replace this with `// Handle ...`

File: server/session/session.go [hunk 4]
```diff
@@ -79,6 +79,10 @@ type Session struct {
 	openChunkTransactions []map[uint64]struct{}
 	invOpened             bool
 
+	shouldSendSettingsResponse bool
+	serverSettingsResponse     []byte
+	serverSettingsFormID       uint32
```
Comment (JustTalDevelops) [inline server/session/session.go:84]: You don't ever update this field.

File: server/session/session.go [hunk 5]
```diff
@@ -79,6 +79,10 @@ type Session struct {
 	openChunkTransactions []map[uint64]struct{}
 	invOpened             bool
 
+	shouldSendSettingsResponse bool
```
Comment (JustTalDevelops) [inline server/session/session.go:82]: I would rename this and the field below it to something slightly better, maybe `hasServerSettings` and `serverSettings`

Comment (JustTalDevelops) [reply server/session/session.go:82]: These should also probably be atomic

File: server/session/session.go (hunk 4 above)
Comment (JustTalDevelops) [reply server/session/session.go:84]: This should also be atomic, if you intend on using it.

File: server/session/session.go [hunk 6]
```diff
@@ -538,3 +544,14 @@ func (s *Session) sendAvailableEntities(w *world.World) {
 	}
 	s.writePacket(&packet.AvailableActorIdentifiers{SerialisedEntityIdentifiers: serializedEntityData})
 }
+
+// SetServerSettings Sets the server settings data.
+// This will be used when a ServerSettingsRequest packet is
+// sent by the client.
+func (s *Session) SetServerSettings(form form.Form) {
```
Comment (JustTalDevelops) [inline server/session/session.go:551]: The documentation here is poor. You should look at the Go documentation style guide: https://go.dev/blog/godoc

**Review by TacoError — COMMENTED** [review]
sasad

**Comment by Sandertv** [comment]
This method isn't actually accessible to users as access to sessions is restricted. How do you reckon we should expose this to users?

**Comment by TacoError** [comment]
Didn't mean to close this, purged my github repositories. My bad.

**Pull request closed without merging**